### ISSUE TITLE
Add adaptive sampler

### ISSUE THREAD

**Issue by adriancole** [openzipkin/zipkin-java#164]
Although not widely used, zipkin has historically included an adaptive sampler which can help prevent a surge in traffic from overwhelming the zipkin storage layer. This works in scenarios where you can coordinate the collection/ingestion tier and are running zookeeper.

**Comment by adriancole** [comment]
dropped the separate repo in favor of https://github.com/openzipkin/zipkin-java/pull/162

**Comment by adriancole** [comment]
Shading the code from zipkin-scala isn't going to work. Shade problems are numerous due to scala and twitter.util types having lots of initializers that aren't mapping well. Even if we could get shade working properly, it would result in an 8MB jar and a dependency on https://maven.twttr.com/, which routinely breaks builds.

The current idea is to merge the adapter that uses zipkin-scala code for now. This way, people can start testing it eventhough it has a huge dependency tree.

We'll need to rewrite the entire thing before 1.0 is out, as I don't feel comfortable setting up a dead repo, or putting such a liability in the codebase.

**Comment by adriancole** [comment]
ps here's where I got with shade. problem is that there's some goofy date lookups in the Var (weird observer thing) that break the build.
```xml
  <build>
    <plugins>
      <!-- Scrub deps from legacy zipkin project at the expense of a big jar (esp bc scala) -->
      <plugin>
        <artifactId>maven-shade-plugin</artifactId>
        <executions>
          <execution>
            <phase>package</phase>
            <goals>
              <goal>shade</goal>
            </goals>
            <configuration>
              <shadeTestJar>false</shadeTestJar>
              <minimizeJar>true</minimizeJar>
              <artifactSet>
                <excludes>
                  <exclude>${project.groupId}:zipkin</exclude>
                  <!-- Don't shade zookeeper's direct dependencies -->
                  <exclude>org.apache.zookeeper:zookeeper:*</exclude>
                  <exclude>org.slf4j:slf4j-api:*</exclude>
                  <exclude>org.slf4j:slf4j-log4j12:*</exclude>
                  <exclude>log4j:log4j:*</exclude>
                  <exclude>jline:jline:*</exclude>
                  <exclude>io.netty:netty:*</exclude>
                </excludes>
              </artifactSet>
              <relocations>
                <relocation>
                  <shadedPattern>zipkin.internal.</shadedPattern>
                  <excludes>
                    <!-- Don't repackage zipkin.internal! -->
                    <exclude>zipkin.internal.**</exclude>
                    <!-- Exclude ZooKeeperSampler and its imports -->
                    <exclude>zipkin.sampler.zookeeper.ZooKeeperSampler</exclude>
                    <exclude>zipkin.sampler.zookeeper.ZooKeeperSampler$Builder</exclude>
                    <exclude>zipkin.Sampler</exclude>
                    <exclude>zipkin.Span</exclude>
                    <exclude>java.**</exclude>
                    <!-- Exclude things finagle needs -->
                    <exclude>Scala-Compiler-Version</exclude>
                    <exclude>sun.misc.Unsafe</exclude>
                    <exclude>theUnsafe</exclude>
                    <exclude>state</exclude>
                    <!-- Exclude direct dependencies -->
                    <exclude>org.apache.zookeeper.**</exclude>
                    <exclude>org.jboss.netty.**</exclude>
                    <exclude>org.slf4j.**</exclude>
                    <exclude>org.junit.**</exclude>
                  </excludes>
                </relocation>
              </relocations>
            </configuration>
          </execution>
        </executions>
      </plugin>
    </plugins>
  </build>
```

**Comment by adriancole** [comment]
closed via #167 released as 0.14.0
